library/Datastore.nucleus.js

```diff
@@ -920,8 +924,11 @@ class NucleusDatastore {
               } else {
                 // If the property is a number, collapse as an array.
                 if (nucleusValidator.isArray(accumulator)) {
-                  accumulator[propertyName] = value;
-                  accumulator = accumulator.filter(Boolean);
+                  // if Array is empty
+                  if(value) {
+                    accumulator[propertyName] = value;
+                    accumulator = accumulator.filter(Boolean);
+                  }
```

Comment:
It's needed to don't have such `array[null]' or `array['undefined]` output.
So we just skip invalid values

Reply:
@sebastienfilion Yeah, it will be better